src/open_samus_returns_rando/specific_patches/door_patches.py

```diff
@@ -187,9 +188,153 @@ def _patch_one_way_doors(editor: PatcherEditor):
             properties.components[0]["arguments"][2]["value"] = False
 
 
-def patch_doors(editor: PatcherEditor):
+class ActorData(Enum):
+    """
+    Enum containing data on actors
+
+    actordef: the actordef for the actor
+    minimapData: the MinimapIconData for this type
+    """
+    DOOR_POWER = (["doorpowerpower", "doorpowerclosed", "doorclosedpower"])
+    DOOR_CHARGE = (["doorchargecharge", "doorchargeclosed", "doorclosedcharge"])
+    SHIELD_WAVE_BEAM = (["doorwave"])
+    SHIELD_SPAZER_BEAM = (["doorspazerbeam"])
+    SHIELD_PLASMA_BEAM = (["doorcreature"])
+    SHIELD_MISSILE = (["doorshieldmissile"])
+    SHIELD_SUPER_MISSILE = (["doorshieldsupermissile"])
+    SHIELD_POWER_BOMB = (["doorshieldpowerbomb"])
+
+class DoorType(Enum):
+    """
+    Enum containing info on each door type
+
+    type: the name Randovania calls the door
+    door: the door's ActorData
+    need_shield: whether the actor needs a shield
+    shield: the shield's ActorData
+    """
+    POWER = ("power_beam", ActorData.DOOR_POWER)
+    CHARGE = ("charge_beam", ActorData.DOOR_CHARGE, False, None, [
+        "actors/props/doorchargecharge", "actors/props/door/fx", "sounds/props/doorchargecharge"
+    ])
+    WAVE_BEAM = ("wave_beam", ActorData.DOOR_POWER, True,
+                 ActorData.SHIELD_WAVE_BEAM,["actors/props/doorwave"])
+    SPAZER_BEAM = ("spazer_beam", ActorData.DOOR_POWER, True,
+                    ActorData.SHIELD_SPAZER_BEAM, ["actors/props/doorspazerbeam"])
+    PLASMA_BEAM = ("plasma_beam", ActorData.DOOR_POWER, True,
+                    ActorData.SHIELD_PLASMA_BEAM, ["actors/props/doorcreature"])
+    MISSILE = ("missile", ActorData.DOOR_POWER, True, ActorData.SHIELD_MISSILE, [
+        "actors/props/doorshield", "actors/props/doorshieldmissile"
+    ])
+    SUPER_MISSILE = ("super_missile", ActorData.DOOR_POWER, True, ActorData.SHIELD_SUPER_MISSILE, [
+        "actors/props/doorshield", "actors/props/doorshieldsupermissile"
+    ])
+    POWER_BOMB = ("power_bomb", ActorData.DOOR_POWER, True, ActorData.SHIELD_POWER_BOMB, [
+        "actors/props/doorshield", "actors/props/doorshieldpowerbomb"
+    ])
```

Comment:
I assume with the proper dependency assurance, the new shields play their expected sounds?

Reply:
That would require me to unmute the game 😮.
Added sound dependencies.
Do SM and PB doors actually make any sound? I mean, I see that there is a file for it, i played the file on my pc by opening the bcwav but if I go to a SM or PB door in the game (without any door patches!) and can't hear anything.
Or is this something since "ever" with the rando / duplicated shields?

Reply:
Yes they make sounds :p

Reply:
What about the second part? `Or is this something since "ever" with the rando / duplicated shields?` 
Because I stay right in front of this door and there is nothing 
![image](https://github.com/randovania/open-samus-returns-rando/assets/117127188/79660b6c-fbc6-4c43-a1ff-295adbffc2fe)

Reply:
I've definitely heard sounds play on the rotated doors. I'm not sure why it wouldn't play at times. Too many sounds maybe? Rotation shouldn't matter since I always hear the Missile shields just fine.